database/postgresql/src/main/java/com/example/javaopencommerce/item/PsqlItemRepositoryImpl.java

```diff
@@ -10,121 +10,121 @@
-                        item.getVat(),
-                        item.getImage().getId())
-                ).map(rs -> {
-                    if (CommonRow.isRowSetEmpty(rs)) {
-                        return ItemEntity.builder().build();
-                    }
-                    return this.itemMapper.rowToItem(rs.iterator().next());
-                });
-    }
-
-    @Override
-    public Uni<ItemDetailsEntity> saveItemDetails(ItemDetailsEntity itemDetails) {
-        return this.client.preparedQuery("INSERT INTO ITEMDETAILS (description, lang, name, item_id) " +
-                        "VALUES($1, $2, $3, $4)")
-                .execute(Tuple.of(
-                        itemDetails.getDescription(),
-                        itemDetails.getLang().toLanguageTag(),
-                        itemDetails.getName(),
-                        itemDetails.getItemId()
-                )).map(rs -> {
-                    if (CommonRow.isRowSetEmpty(rs)) {
-                        return ItemDetailsEntity.builder().build();
-                    }
-                    return this.itemMapper.rowToItemDetails(rs.iterator().next());
-                });
-    }
-
-    @Override
-    public Uni<Integer> getItemStock(Long id) {
-        return this.client.preparedQuery("SELECT stock FROM ITEM WHERE id = $1")
-                .execute(Tuple.of(id))
-                .map(rs -> {
-                    if (CommonRow.isRowSetEmpty(rs)) {
-                        return -1;
-                    }
-                    return rs.iterator().next().getInteger("stock");
-                });
-    }
-
-    @Override
-    public Uni<Integer> changeItemStock(Long id, int stock) {
-        return this.client.preparedQuery("UPDATE ITEM SET stock = $1 WHERE id = $2 RETURNING stock")
-                .execute(Tuple.of(stock, id))
-                .map(rs -> {
-                    if (CommonRow.isRowSetEmpty(rs)) {
-                        return -1;
-                    }
-                    return rs.iterator().next().getInteger("stock");
-                });
-    }
+  private final PgPool client;
+  private final ItemMapper itemMapper;
+
+  private static final String SELECT_ITEM_BASE = "SELECT * FROM ITEM i";
+  private static final String SELECT_DETAILS_BASE = "SELECT * FROM ITEM_DETAILS i";
+  private static final String IMAGE_JOIN = "INNER JOIN Image img ON i.image_id = img.id";
+
+  PsqlItemRepositoryImpl(PgPool client) {
+    this.client = client;
+    this.itemMapper = new ItemMapper();
+  }
+
+  @Override
+  public Uni<List<ItemEntity>> getAllItems() {
+    return this.client.preparedQuery(format("%s %s", SELECT_ITEM_BASE, IMAGE_JOIN))
+        .execute()
+        .map(this.itemMapper::getItems);
+  }
+
+  @Override
+  public Uni<ItemEntity> getItemById(Long id) {
+    return this.client.preparedQuery(format("%s %s WHERE i.id = $1", SELECT_ITEM_BASE, IMAGE_JOIN))
+        .execute(Tuple.of(id))
+        .map(rs -> {
+          if (CommonRow.isRowSetEmpty(rs)) {
```

Comment:
There's no `.filter().orElse()` ? 
And maybe we shouldn't return empty ItemEntity, but empty Uni instead?

Reply:
Keeping as an issue, not connected with this PR